dask/dataframe/io/tests/test_csv.py

```diff
@@ -1666,3 +1666,37 @@ def test_csv_parse_fail(tmpdir):
         dd.read_csv(path, sample=13)
     df = dd.read_csv(path, sample=13, sample_rows=1)
     assert_eq(df, expected)
+
+
+def test_csv_name_should_be_different_even_if_head_is_same(tmpdir):
+    # https://github.com/dask/dask/issues/7904
+    import random
+    from shutil import copyfile
+
+    old_csv_path = os.path.join(str(tmpdir), "old.csv")
+    new_csv_path = os.path.join(str(tmpdir), "new_csv")
+
+    # Create random CSV
+    with open(old_csv_path, "w") as f:
+        for _ in range(10):
+            f.write(
+                f"{random.randrange(1, 10**9):09}, {random.randrange(1, 10**9):09}, {random.randrange(1, 10**9):09}\n"
+            )
+
+    copyfile(old_csv_path, new_csv_path)
+
+    # Add three new rows
+    with open(new_csv_path, "a") as f:
+        for _ in range(3):
+            f.write(
+                f"{random.randrange(1, 10**9):09}, {random.randrange(1, 10**9):09}, {random.randrange(1, 10**9):09}\n"
+            )
+
+    new_df = dd.read_csv(
+        "new.csv", header=None, delimiter=",", dtype=str, blocksize=None
+    )
+    old_df = dd.read_csv(
+        "old.csv", header=None, delimiter=",", dtype=str, blocksize=None
+    )
```

Comment:
It looks like we should use `old_csv_path` / `new_csv_path` instead of `"old.csv"` / `"new.csv"`, respectively

Reply:
:facepalm: yep!